ReadMe-FormVer.md

```diff
@@ -12,8 +12,87 @@ dist/kotlinc/bin/kotlinc -language-version 2.0 -Xcompiler-plugin=dist/kotlinc/li
 
 ## Viper dependency
 
-TODO: Viper may need to be available locally to build the
-project, a HOWTO would be nice to have.
+To build the plugin, two main Viper dependencies are needed: 
+[silver](https://github.com/viperproject/silicon) (the API for
+generation the intermediate representation) and 
+[silicon](https://github.com/viperproject/silicon) (the symbolic engine for
+performing the verification of Viper code).
+
+However, the `silicon` *project structure* and *building file* require some
+important modifications for allowing Gradle to resolve the dependencies required by both projects.
+
+Therefore, to install the dependencies, be sure to have installed locally:
+* The Java Development Kit
+* [Maven](https://maven.apache.org/index.html)
+* [SBT](https://www.scala-sbt.org/)
+* [Z3](https://github.com/Z3Prover/z3) v4.8.7
+
+The installing of `Z3` is not necessary enough, it is important to have the JAR dependency required by `silicon`.
+Thus, open your terminal emulator and download the file (you can use `curl`/`wget`):
+
+```bash
+cd /tmp
+# Download the Z3 jar file from the following provider (used by silicon)
+wget https://www.sosy-lab.org/ivy/org.sosy_lab/javasmt-solver-z3/com.microsoft.z3-4.8.7.jar
+# Install the Z3 jar file into the local Maven repository
+mvn install:install-file \
+   -Dfile=./com.microsoft.z3-4.8.7.jar \
+   -DgroupId=com.microsoft \
+   -DartifactId=z3 \
+   -Dversion=4.8.7 \
+   -Dpackaging=jar \
+   -DgeneratePom=true
+```
+
+Now, we can clone `silicon` from its GitHub repository.Once the cloning process is complete, open `silicon`'s `build.sbt` in your favorite editor
+and modify it applying the following [patch](./resources/patches/silicon-edits.patch).
+
+```bash
+# Clone the Silicon project in a directory.
+# The flag `--recursive` will also clone `silver`.
+git clone --recursive https://github.com/viperproject/silicon
+
+cd silicon
+
+# Apply the patch with the needed modifications
+git apply silicon-edits.patch
+```
+
+
+Now, open up the `silver`'s builiding file (located at `./silicon/silver/build.sbt`), and in the _Publishing settings_, 
+add the following lines:
+
+```sbt
+// Publishing settings
+/* ... */
+ThisBuild / publishArtifact := true
+publishTo := Some(MavenCache("local-maven", file(Path.userHome.absolutePath + "/.m2/repository")))
+```
+
+Once done with all the modifications, from the `silicon` root directory we can compile and publish the project
+running `sbt`:
+
+```bash
+sbt compile
+sbt publish
```

Comment:
At this point I only have `~/.m2/repository/viper/silicon_2.13`, no corresponding directory for silver; I think you need to do the same in the `silver` directory, or maybe need to pass some kind of recursive flag?

Reply:
That's strange because, did you apply the `ThisBuild / publishArtifact := true` flag inside `silver`'s build file?

Reply:
Yes, I did.  I think running `sbt compile` in `silicon` didn't build Silver at all.